src/words/tweet_tokenizer.jl

```diff
@@ -0,0 +1,953 @@
+        !isempty(entity) && push!(ts.buffer, entity[1])
+        ts.idx = i + 1
+        return true
+    else
+        number = -1
+        i = ts.idx + 2
+        if ts.input[ts.idx + 2] != 'x'    # Entity is of the type "&#916;" => "Δ"
+            while i <= length(ts.input) && isdigit(ts[i])
+                i += 1
+            end
+            (i > length(ts.input) || ts[i] != ';') && return false
+            if ((ts.idx + 2 ) == i)
+                !remove_illegal && return false
+                ts.idx +=3
+                return true
+            end
+            (number = parse(Int, String(ts[ts.idx+2:i-1]), base=10))
+        else   # Entity is of the type "&#x394;" => "Δ"
+            i += 1
+            base16letters = ('a', 'b', 'c', 'd', 'e', 'f')
+            while i <= length(ts.input) && (isdigit(ts[i]) || ts[i] in base16letters)
+                i += 1
+            end
+            (i > length(ts.input) || ts[i] != ';') && return false
+
+            if (ts.idx + 3) == i
+                !remove_illegal && return false
+                ts.idx += 4
+                return true
+            end
+            number = parse(Int, String(ts[ts.idx+3:i-1]), base=16)
+        end
+
+        windows_1252_chars = ['€', '\u81', '‚', 'ƒ', '„', '…', '†', '‡', 'ˆ', '‰',
+                          'Š', '‹', 'Œ', '\u8d','Ž', '\u8f', '\u90', '‘', '’',
+                          '“', '”', '•', '–', '—', '˜', '™', 'š', '›', 'œ',
+                          '\u9d', 'ž', 'Ÿ']
+        if 0x80 <= number <= 0x9F
+            push!(ts.buffer, windows_1252_chars[number - 127])
+            ts.idx = i + 1
+            return true
+        end
+        if (number <= 0 || !Unicode.isassigned(number))
+            !remove_illegal && return false
+            ts.idx = i + 1
+        else
+            push!(ts.buffer, Char(number))
+        ts.idx = i + 1
+        end
+    end
+    return true
+end
+
+"""
+    lookbehind(ts::TokenBuffer)
+A helper function for strip_twitter_handle. Checks if the beginning of the detected
+handle is preceded by alphanumeric or special chars like('_', '!', '@', '#', '\$', '%', '&', '*')
+"""
+function lookbehind(ts::TokenBuffer,
+                match_pattern = ('_', '!', '@', '#', '$', '%', '&', '*'))
+    ts.idx == 1 && return false
+
+    c = ts[ts.idx - 1]
+    ( islowercase(c) || isdigit(c) || isuppercase(c) || c ∈ match_pattern ) && return true
+
+    return false
+end
+
+
+"""
+    strip_twitter_handle(ts::TokenBuffer)
+
+For removing Twitter Handles. If it detects a twitter handle, then it jumps to
+makes the index of TokenBuffer to the desired location skipping the handle.
+"""
+function strip_twitter_handle(ts)
+
+    (ts.idx + 2 > length(ts.input) || ts.input[ts.idx] != '@' ) && return false
+    lookbehind(ts) && return false
+
+    i = ts.idx + 1
+    while i <= length(ts.input) &&
+              ( isascii(ts[i]) && (isdigit(ts[i]) || islowercase(ts[i]) ||
+               isuppercase(ts[i]) || ts[i] == '_'))
```

Comment:
Can this be tidied up somewhat?
For a start `isalpha` rather than checking if uppercase or lower case.
but also break the lines up differently to make it clearer what is part of the condition.

One option is
```
while (
    i <= length(ts.input)
    && isascii(ts[i])
    && (isdigit(ts[i]) || isalpha(ts[i]) || ts[i] == '_')
)
```

another option is:
```
while i <= length(ts.input)
    isascii(ts[i]) || break
    isdigit(ts[i]) || isalpha(ts[i]) || ts[i] == '_' || break
```

Reply:
Sure. I will also do so in other functions.
How's the idea of the following one line functions in tweet_tokenizer.jl
`isalpha(c) = isascii(c) && (islowercase(c) || isuppercase(c))`
and 
`isalphanum(c) = isascii(c) && (islowercase(c) || isuppercase(c) || isdigit(c))`

Reply:
Please check now.